src/platform/Darwin/DnssdImpl.cpp

```diff
@@ -185,15 +278,18 @@ CHIP_ERROR Register(void * context, DnssdPublishCallback callback, uint32_t inte
         return CHIP_NO_ERROR;
     }
 
-    sdCtx = chip::Platform::New<RegisterContext>(type, callback, context);
-    VerifyOrReturnError(nullptr != sdCtx, CHIP_ERROR_NO_MEMORY);
-
-    DNSServiceRef sdRef;
-    auto err = DNSServiceRegister(&sdRef, kRegisterFlags, interfaceId, name, type, kLocalDot, nullptr, ntohs(port), record.size(),
-                                  record.data(), OnRegister, sdCtx);
-    VerifyOrReturnError(kDNSServiceErr_NoError == err, sdCtx->Finalize(err));
+    std::string hostnameStr                   = std::string(hostname) + '.' + kLocalDot;
```

Comment:
Can you add a method that does that ? 
```
std::string GetHostnameWithDomain(const char * hostname)
{
    return std::string(hostname) + '.' + kLocalDot;
}
```

It makes it clear what we are doing (and there is an other `GetHostnameWithoutDomain` method living in `DnsContexts.cpp`

Also can you move the hostname assignment a bit later (after `VerifyOrReturnError(nullptr != registerRecordCtx, CHIP_ERROR_NO_MEMORY);`) since there is no need to create this variable  if we are going to return early.

Reply:
makes sense.

Reply:
done